wolfcrypt/src/asn.c

```diff
@@ -9117,64 +9117,54 @@ int wc_EccPublicKeyDecode(const byte* input, word32* inOutIdx,
                           ecc_key* key, word32 inSz)
 {
     int    length;
-    int    ret = 0;
+    byte   b;
 
     if (input == NULL || inOutIdx == NULL || key == NULL || inSz == 0)
         return BAD_FUNC_ARG;
 
     if (GetSequence(input, inOutIdx, &length, inSz) < 0)
         return ASN_PARSE_E;
 
-#if defined(OPENSSL_EXTRA) || defined(ECC_DECODE_EXTRA)
-    {
-        byte b = input[*inOutIdx];
-        if (b != ASN_INTEGER) {
-            /* not from decoded cert, will have algo id, skip past */
-            if (GetSequence(input, inOutIdx, &length, inSz) < 0)
-                return ASN_PARSE_E;
+    if (GetSequence(input, inOutIdx, &length, inSz) < 0)
```

Comment:
The only call I can see to wc_EccPublicKeyDecode() in the library is in wc_SetSubjectKeyId().

The format of the public ECC key ./certs/ecc-client-keyPub.pem and ./certs/ecc-keyPub.der (used in wolfcrypt/test/test.c) is:
    0:d=0  hl=2 l=  89 cons: SEQUENCE          
    2:d=1  hl=2 l=  19 cons:  SEQUENCE          
    4:d=2  hl=2 l=   7 prim:   OBJECT            :id-ecPublicKey
   13:d=2  hl=2 l=   8 prim:   OBJECT            :prime256v1
   23:d=1  hl=2 l=  66 prim:  BIT STRING        

The format of the public key in the certificate client-ecc-cert.pem is:
  356:d=2  hl=2 l=  89 cons:   SEQUENCE          
  358:d=3  hl=2 l=  19 cons:    SEQUENCE          
  360:d=4  hl=2 l=   7 prim:     OBJECT            :id-ecPublicKey
  369:d=4  hl=2 l=   8 prim:     OBJECT            :prime256v1
  379:d=3  hl=2 l=  66 prim:    BIT STRING        

The certificate parsing code pulls apart the key manually in GetKey().

I can find no data or call that attempts to use wc_EccPublicKeyDecode() with anything other than the above format. I'll push (force) an update with the rebase to master.